examples/lighting-app/silabs/src/ColorFormat.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ColorFormat.h"
+
+#include <math.h>
+
+// define a clamp macro to substitute the std::clamp macro which is available from C++17 onwards
+#define clamp(a, min, max) ((a) < (min) ? (min) : ((a) > (max) ? (max) : (a)))
+
+RgbColor_t HsvToRgb(HsvColor_t hsv)
+{
+    RgbColor_t rgb;
+
+    uint8_t region, p, q, t;
+    uint32_t h, s, v, remainder;
+
+    if (hsv.s == 0)
+    {
+        rgb.r = rgb.g = rgb.b = hsv.v;
+    }
+    else
+    {
+        h = hsv.h;
+        s = hsv.s;
+        v = hsv.v;
+
+        region    = h / 43;
+        remainder = (h - (region * 43)) * 6;
+        p         = (v * (255 - s)) >> 8;
+        q         = (v * (255 - ((s * remainder) >> 8))) >> 8;
+        t         = (v * (255 - ((s * (255 - remainder)) >> 8))) >> 8;
+        switch (region)
```

Comment:
need code comments to explain what is done and what are regions

Reply:
Added comments and added utility functions for calculations and convert functions